src/generic/listctrl.cpp

```diff
@@ -2391,6 +2391,9 @@ wxTextCtrl *wxListMainWindow::EditLabel(long item, wxClassInfo* textControlClass
     wxASSERT_MSG( textControlClass->IsKindOf(wxCLASSINFO(wxTextCtrl)),
                  wxT("EditLabel() needs a text control") );
 
+    wxTextCtrl * const text = (wxTextCtrl *)textControlClass->CreateObject();
+    m_textctrlWrapper = new wxListTextCtrlWrapper(this, text, item);
+
```

Comment:
Can we move these lines immediately before `ProcessEvent()` call? This would avoid the problem with orphan text control being left if `wxCHECK_MSG` below fails and would also make it more clear that we need to create the control before calling the handler.